reference/vane-apis/ames.md

```diff
@@ -0,0 +1,303 @@
+
+A vane can pass Ames a `%heed` `task` to request Ames track a peer's
+responsiveness.  If our `%boon`s to it start backing up locally,
+Ames will `give` a `%clog` back to the requesting vane containing the
+unresponsive peer's Urbit address.
+
+Stop tracking a peer by sending Ames a
+`%jilt` `task`.
+
+
+#### Accepts
+
+```hoon
+=ship
+```
+
+The ship to be tracked.
+
+#### Returns
+
+The `+on-heed` arm returns `event-core` with `heeds` modified to include a
+`duct` that points to the input `ship`. (this isn't a `gift` so maybe I should
+not include this?)
+
+If the `ship` is indeed being unresponsive, as measured by backed up `%boon`s,
+Ames will `give` a `%clog` `gift` to the requesting vane containing the
+unresponsive peer's urbit address.
+
+
+### %hole
+
+`%hole` handles packet crash notification. `%hole` works much like `%hear`, in
+that it passes an incoming raw packet to `+decode-packet` to be deserialized,
+and then passes that data along with the source of the packet to
+`+on-hear-packet` along with a `?` set to `%.n` denoting that there is something wrong with
+the packet. 
+
+#### Accepts
+
+```hoon
+[=lane =blob]
+```
+`%hole` takes in a `blob`, which is essentially a large atom (around 1kB or less)
+that is the raw data of the message, and a `lane`, which is the origin of the
+message (typically an IP address).
+
+#### Returns
+
+Same confusion as with `%hear`.
+
+
+### %jilt
+
+`%jilt` stops tracking a potentially unresponsive peer that was previously being
+tracked as a result of the `%heed` `task`.
+
+There are two `+on-jilt` arms, this `task` utilizes `on-hear:event-core`.
+
+#### Accepts
+
+```hoon
+=ship
+```
+
+The `ship` we no longer wish to track.
+
+#### Returns
+
+`+on-jilt` returns `event-core` with the `duct` associated to `ship` removed from `heeds`, assuming it is
+there. Otherwise it returns `event-core` unchanged.
+
+Does `%jilt` return any gifts?
+
+
+### %plea
+
+`%plea` is the `task` used to send messages over Ames. It extends the
+`%pass`/`%give` semantics across the network. As such, it is the most
+fundamental `task` in Ames and the primary reason for its existence.
+
+Ames `pass`es a `%plea` `note` to another vane when it receives a message on a
+"forward flow" from a peer, originally passed from one of the peer's vanes to
+peer's Ames.
+
+Ames `pass`es a `%plea` to itself to trigger a heartbeat message to be sent to
+our ship's sponsor
+
+#### Accepts
+
+```hoon
+[=ship =plea]
+```
+
+A `%plea` `task` takes in the `ship` the `plea` is addressed to, and a `$plea`,
+which is
+
+```hoon
+  +$  plea  [vane=@tas =path payload=*]
+```
+Here, `vane` is the destination vane on the remote ship, `path` is the internal
+route on the receiving ship, and `payload` is the semantic message content.
+
+#### Returns
+
+`event-core` is returned, modified to include the received `%plea` (this is not
+a `gift`).
+
+Not sure a `gift` is returned? But if there would be an ack `gift` anywhere, I
+feel like it would be here.
```

Comment:
No returns to the caller.  This may trigger more messages to be given to unix, but that depends on congestion control (how many packets are in flight, etc).